### PR TITLE
Irc sasl

### PR DESCRIPTION
**Pull request by keur**
We deliberately do not support the dh_aes (or its predecessor dh_blowfish)
mechanisms. These mechanisms are for users that want to send their
username and passwords encrypted to the server over an insecure
connection. Since the OCF only supports connecting over TLS there is no
point.

### CODE REVIEW

**Comment by keur** [comment]
@jvperrin I've tested it on dev-flood. SASL PLAIN is the recommended way to connect to connect to IRC these days. It registers you with the server before the connection is made.

**Comment by jvperrin** [comment]
Ah I see, carry on then, looks great!

**Comment by dkess** [comment]
Can you please explain the PID file changes you made?

**Comment by keur** [comment]
@dkess I moved the location from `/var/run` to `/run`; the latter is mounted as tmpfs and nowadays more standard for storing runtime information. The PIDFile= tells systemd where to look for the PID file and uses that to set $MAINPID. It's not strictly necessary since we specified the service as non-forking, but is good practice to use in services that create PID files (anope will regardless). The PermissionsStartOnly= directive is deprecated so I removed it since the PID file is being created with the correct user/group.

File: modules/ocf_irc/templates/inspircd.conf.erb [hunk 1]
```diff
@@ -13,6 +13,7 @@
 <module name="m_sha256.so">
 <module name="m_sqloper.so">
 <module name="m_ssl_gnutls.so">
+<module name="m_cap.so">
```
Comment (cg505) [inline modules/ocf_irc/templates/inspircd.conf.erb]: why was this added?

Comment (ja5087) [reply modules/ocf_irc/templates/inspircd.conf.erb]: Seems to be the capability negotiation module. I was wondering about it too.

Comment (keur) [reply modules/ocf_irc/templates/inspircd.conf.erb]: SASL won't work otherwise. It's part of the ircv3 spec (SASL is ircv3, although everyone supports it). Your client will do use it to see available ircv3 features the server supports. You can do this in your client with `/quote CAP LS`

https://ircv3.net/specs/extensions/sasl-3.2

Comment (cg505) [reply modules/ocf_irc/templates/inspircd.conf.erb]: Awesome, sorry I missed that.
